From 6c96210583e788d8929deaccbaa5aa6e5fd7bc4a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cl=C3=A9ment=20Le=20Goffic?= <clement.legoffic@foss.st.com>
Date: Fri, 23 Feb 2024 15:23:38 +0100
Subject: [PATCH 1149/1695] dts: stm32: add wakeup properties to the RTC
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The 'wakeup-source' property allows to say that the RTC peripheral
can wakeup the platform.
Overload the property 'wakeup-parent' for the RTC as the wakeup-parent
capable of wakeup the platform with a RTC signal input is the exti2.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 0da3ee47c1dbc9106a8ddeb804cff7cf27c78dce dts: stm32: add wakeup properties to the RTC for stm32mp21
844de6520fe389fd03766d78b62f28eea3b0414f dts: stm32: add wakeup properties to the RTC for stm32mp23 plateform
579a367d92826df927c30f8406ad6957f84512f8 dts: stm32: add wakeup properties to the RTC]
Change-Id: I81e4566e165dc61a4efaeab28d95aca18a31cf18
Signed-off-by: Clément Le Goffic <clement.legoffic@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/402363
Domain-Review: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 9176ea25f..4cfb9c7ad 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -1219,6 +1219,8 @@
 			reg = <0x46000000 0x400>;
 			clocks = <&rcc CK_BUS_RTC>, <&rcc RTC_CK>;
 			clock-names = "pclk", "rtc_ck";
+			wakeup-source;
+			wakeup-parent = <&exti2>;
 			interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL_HIGH>;
 		};
 
-- 
2.47.2

